overlord/devicestate/devicemgr.go

```diff
@@ -885,10 +885,30 @@ func (m *DeviceManager) RequestSystemAction(systemLabel string, action SystemAct
 		return ErrUnsupportedAction
 	}
 
-	// TODO:UC20 assume we are in the run mode
-	if sysAction.Mode == "run" {
-		// do nothing
-		return nil
+	switch m.systemMode {
+	case "recover", "run":
+		// if going from recover to recover or from run to run and the systems
+		// are the same do nothing
+		if m.systemMode == sysAction.Mode && systemLabel == currentSys.System {
```

Comment:
Unless there would be a multi-reboot recovery process, but I guess if one can call this API, one could also issue a reboot directly.

Reply:
yes I don't think we want workflows that need to reboot for $SOMETHING to be using snapd to trigger this reboot, because as far as snapd is concerned, it doesn't need to _do anything_ to get to run mode when we're already in run mode. And yes such management agents should already have root to call this endpoint, so they probably can just reboot themselves